core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -2634,6 +2634,127 @@ private static String getString(ImmutableMap<RexNode, RexNode> map) {
     assertThat(simplifiedExpr, is(origExpr));
   }
 
+  @Test void testSimplifyInValues() {
+    RexNode a = vIntNotNull(1);
+    RexNode literal1 = rexBuilder.makeLiteral(1, a.getType(), true);
+    RexNode literal2 = rexBuilder.makeLiteral(2, a.getType(), true);
+    RexNode literal3 = rexBuilder.makeLiteral(3, a.getType(), true);
+    RexNode literal4 = rexBuilder.makeLiteral(4, a.getType(), true);
+    RexNode literal5 = rexBuilder.makeLiteral(5, a.getType(), true);
+    RexNode literal6 = rexBuilder.makeLiteral(6, a.getType(), true);
+
+    // expr: a in (1, 2, 3, 4, 5)
+    RexNode expr = rexBuilder.makeCall(
+        SqlStdOperatorTable.IN, a, literal1, literal2, literal3, literal4, literal5);
+    // expected: a >= 1 and a <= 5
+    RexNode expected = rexBuilder.makeCall(SqlStdOperatorTable.AND,
+        rexBuilder.makeCall(SqlStdOperatorTable.GREATER_THAN_OR_EQUAL, a, literal1),
+        rexBuilder.makeCall(SqlStdOperatorTable.LESS_THAN_OR_EQUAL, a, literal5));
```

Comment:
Please use `RexProgramTestBase` DSL when writing tests (`and`, `or` and similar methods instead of `makeCall` and similar `rexBuilder`)

Reply:
@vlsi Revised accordingly, and the tests look much clearer. Thanks a lot for your good suggestion.